packages/core/src/rum/instrumentation/resourceTracking/implementation/DatadogRumResource/ResourceReporter.ts

```diff
@@ -0,0 +1,78 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+import { DdRum } from '../../../../../foundation';
+import type { RUMResource } from '../../domain/interfaces/RumResource';
+
+import { createTimings } from './resourceTiming';
+
+type ResourceMapper = (resource: RUMResource) => RUMResource | null;
```

Comment:
If we are going to expose it in the public API, I would propose to use another type for the input argument here. In both iOS and Android SDKs during the event mapping phase we allow only to modify certain parts of the events, so that event data which is used in visualisation, some metrics calculation, etc. is not modified by the user code (otherwise it may break the integrity/validity of the data).

So I think here we may probably allow to modify URL and that is it (it may be useful in case of REST APIs, where, say, some resource IDs are part of the URL, but in the end it is the same endpoint, so such URLs should be grouped), I don't think we need to allow modification of the timings, `key`, `kind`, `method`, etc. properties.

If it won't be a part of public API it is good as it is.